commands/FBClassDump.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+
+import os
+import re
+import string
+
+import lldb
+import fblldbbase as fb
+import fblldbobjcruntimehelpers as runtimeHelpers
+
+def lldbcommands():
+  return [
+    FBPrintMethods()
+  ]
+
+class FBPrintMethods(fb.FBCommand):
+  def name(self):
+    return 'pmethods'
+
+  def description(self):
+    return 'Print the class instance methods.'
+
+  def options(self):
+    return [
+      fb.FBCommandArgument(short='-a', long='--all', arg='all', help='If display all methos include class and instance methods', default=False, boolean=True),
+      fb.FBCommandArgument(short='-c', long='--class', arg='clsmethod', help='Print the class methods only', default=False, boolean=True)
+    ]
+
+  def args(self):
+    return [ fb.FBCommandArgument(arg='class or instance', type='id or Class', help='an Object-C Class.') ]
+
+  def run(self, arguments, options):
+    cls = arguments[0]
+    if not isClassObject(cls):
+        cls = runtimeHelpers.object_getClass(cls)
+        if not isClassObject(cls):
+            raise Exception('parameter invalide, not a id or Class')
+
+    if options.all:
+        printClassMethods(cls)
+        printInstanceMethods(cls)
+    elif options.clsmethod:
+        printClassMethods(cls)
+    else:
+        printInstanceMethods(cls)
+
+def isClassObject(arg):
+    return runtimeHelpers.class_isMetaClass(runtimeHelpers.object_getClass(arg))
+
+def printInstanceMethods(cls, prefix='-'):
+    ocarray = instanceMethosOfClass(cls)
+    if not ocarray:
+      print "-- have none method or an error occur-- "
+      return
+
+    methodAddrs = covertOCArrayToPyArray(ocarray)
+    methods = []
+    for i in methodAddrs:
+      method = createMethodFromOCMethod(i)
+      if method is not None:
+        methods.append(method)
+        print prefix + ' ' + method.prettyPrint()
+
+def printClassMethods(cls):
+    printInstanceMethods(runtimeHelpers.object_getClass(cls), '+')
+
+# I find that a method that has variable parameter can not b.evaluateExpression
+# so I use numberWithLongLong: rather than -[NSString stringWithFormat:]
+def instanceMethosOfClass(klass):
+  tmpString = """
+    unsigned int outCount;
+    void **methods = (void **)class_copyMethodList((Class)$cls, &outCount);
+    NSMutableArray *result = [NSMutableArray array];
+    for (int i = 0; i < outCount; i++) {
+      NSNumber *num = (NSNumber *)[NSNumber numberWithLongLong:(long long)methods[i]];
+      [result addObject:num];
+    }
+    (void)free(methods);
+    id ret = result.count ? [result copy] : nil;
+    ret;
+  """
+
+  command = string.Template(tmpString).substitute(cls=klass)
+  command = '({' + command + '})'
+  ret = fb.evaluateExpressionValue(command)
+  if not ret.GetError().Success():
+    return None
+
+  ret = ret.GetValue()
+  if int(ret, 16) == 0: # return nil
```

Comment:
This could be removed if you use fb.evaluateIntegerExpressionValue a few lines above.